programs/marginfi/src/instructions/marginfi_account/borrow.rs

```diff
@@ -78,7 +85,38 @@ pub fn lending_account_borrow<'info>(
             .transpose()?
             .unwrap_or(amount);
 
-        bank_account.borrow(I80F48::from_num(amount_pre_fee))?;
+        let origination_fee_u64: u64;
+        if !origination_fee_rate.is_zero() {
+            let origination_fee: I80F48 = I80F48::from_num(amount_pre_fee)
+                .checked_mul(origination_fee_rate)
+                .ok_or_else(math_error!())?
+                .floor(); // round fees down in the user's favor
```

Comment:
Floor fee (user's favor) or no rounding (users and liquidators repay rounding up)?

Reply:
I'd avoid flooring the fee to avoid situations where fee is reduced by constructing the borrows in a specific way.
The current accounting and fee model should correctly any fractional parts.

Reply:
Note the changes in tests in latest commit where the rounding of the fee must be accommodated by adding 1 to the repay amt.